packages/devextreme/testing/testcafe/helpers/testAccessibility.ts

```diff
@@ -0,0 +1,94 @@
+import createWidget, { WidgetName } from './createWidget';
+import { a11yCheck, A11yCheckOptions, ElementContext } from './accessibilityUtils';
+import { isMaterialBased } from './themeUtils';
+
+export interface Options {
+  [key: string]: any[];
+}
+
+export interface Configuration {
+  component: WidgetName;
+  options?: Options;
+  a11yCheckConfig?: A11yCheckOptions;
+  selector?: ElementContext;
+  created?: (t?: any, optionConfiguration?: Options) => Promise<void>;
+  after?: (t?: any, optionConfiguration?: Options) => Promise<void>;
+}
+
+const defaultSelector = '#container';
+const defaultOptions = {};
+const defaultCreated = async () => {};
+const defaultAfter = async () => {};
+const defaultA11yCheckConfig = isMaterialBased() ? {
+  runOnly: 'color-contrast',
+} : {};
+
+const generateConfigurations = (
+  options: Options,
+  index = 0,
+  prevConfigurations: Options[] = [],
+) => {
+  const keys = Object.keys(options);
+
+  if (index >= keys.length) {
+    return [Object.assign({}, ...prevConfigurations)];
+  }
+
+  const key = keys[index];
+  const values = options[key];
+
+  let configurations: object[] = [];
+
+  values.forEach((value) => {
+    const currentConfigurations = [
+      ...prevConfigurations,
+      { [key]: value },
+    ];
+
+    const generatedConfigurations = generateConfigurations(
+      options,
+      index + 1,
+      currentConfigurations,
+    );
+
+    configurations = configurations.concat(generatedConfigurations);
```

Comment:
`concat` wastes a linear time. Let's use `push` here
```suggestion
    configurations = configurations.push(...generatedConfigurations);
```

Reply:
Strange to me because we remove concat but add spread + push.

Reply:
spread + push iterates only through `generatedConfigurations` while `concat` recreates the whole array

Reply:
Done